crates/matrix-sdk-ui/tests/integration/timeline/queue.rs

```diff
@@ -228,8 +228,12 @@ async fn clear_with_echoes() {
     // Send a message without mocking the server response.
     timeline.send(RoomMessageEventContent::text_plain("Send failure").into(), None).await;
 
-    // Wait for the first message to fail.
-    sleep(Duration::from_millis(10)).await;
+    // Wait for the first message to fail. Don't use time, but listen for the first
+    // timeline item diff to get back signalling the error.
+    {
+        let (_, mut timeline_stream) = timeline.subscribe().await;
```

Comment:
This is racy, the stream should be created before the call to `timeline.send` above.

Reply:
Alternatively, the first tuple element could be not ignored here and checked for the `SendingFailed` local echo state, but that will be more complicated.

Reply:
Is the new version more robust?

Reply:
Yes, that's what I was thinking of! 👍🏼